e2e_test/e2e_test.go

```diff
@@ -69,6 +70,74 @@ var _ = g.Describe("Dpu side", g.Ordered, func() {
 	g.AfterEach(func() {
 	})
 
+	g.It("Should run Webhook to validate DpuOperatorConfig", func() {
+		var err error
+		var config configv1.DpuOperatorConfig
+		var dpuOperatorConfig1 configv1.DpuOperatorConfig
+
+		crNamespace := "openshift-dpu-operator"
+		crName := "dpu-operator-config"
+
+		dpuOperatorConfig0 := configv1.DpuOperatorConfig{}
+
+		// If this failed, do you have a DpuOperatorConfig in your cluster?
+		// Create it with `oc create -f ./examples/dpu.yaml`.
+		Eventually(func() error {
+			return dpuSideClient.Get(context.TODO(), types.NamespacedName{Namespace: crNamespace, Name: crName}, &dpuOperatorConfig0)
+		}, testutils.TestAPITimeout, testutils.TestRetryInterval).Should(Succeed())
+
+		// We delete the DpuOperatorConfig and restore it at the end. This is
+		// dangerous, if the test fails, we end up deleting the
+		// DpuOperatorConfig on the cluster and the user needs to manually fix
+		// it.
+		dpuSideClient.Delete(context.TODO(), &dpuOperatorConfig0)
+
+		dpuOperatorConfig1 = configv1.DpuOperatorConfig{}
+		Eventually(func() error {
+			return dpuSideClient.Get(context.TODO(), types.NamespacedName{Namespace: crNamespace, Name: crName}, &dpuOperatorConfig1)
+		}, testutils.TestAPITimeout, testutils.TestRetryInterval).ShouldNot(Succeed())
+
+		config = configv1.DpuOperatorConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: crNamespace,
+				Name:      "invalidname",
+			},
+			Spec: configv1.DpuOperatorConfigSpec{
+				Mode:     "host",
+				LogLevel: 2,
+			},
+		}
+		err = dpuSideClient.Create(context.TODO(), &config)
+		Expect(err).To(HaveOccurred())
+
+		config = configv1.DpuOperatorConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: crNamespace,
+				Name:      crName,
+			},
+			Spec: configv1.DpuOperatorConfigSpec{
+				Mode:     "invalidmode",
+				LogLevel: 2,
+			},
+		}
```

Comment:
Helper function to create a DpuOperatorConfig CR? You're using it twice here.

Reply:
Are you preparing a follow up PR immediately after this?

Reply:
that is not planned (there is no Jira issue for that).